Simprints/Android-Simprints-ID
# Hotfix sessions add id for realm classes

- Added ids for any nested realm classes - that prevents duplications when we call insertOrUpdate(session)
- Cascade deletion when we do the sync
- Improved tests

If you want to test:
1) set 15m in ScheduledSessionsSyncManager
2) Install the app and set a breakpoint in RealmSessionEventsDbManagerImpl:48
3) Evaluate BigInteger(1, this.localDbKey.value).toString(16)
4) grab the Realm key
5) Make couple of sessions
6) grab the db with 'Device File Explorer'  in data/data/com.simprints.id/files/event_data.realm
7) open the db with 'Realm Browser' using the realm key (point 4)
8) check that sessions and all events are there
9) leave the phone for about 15 minutes
10) Do point 6-7 and check that the db is empty (after the sync we delete the sessions)
